[question] method call hierarchy

Hello,
I'm new to spoon and I'm wondering if following is possible to do with spoon.

I have method `externalServiceCall(...)` which is called from many places in my application. Can I have call hierarchy **to** this method?
What I want to achieve is that I will generate mappings of my [@RestController](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RestController.html) to `externalServiceCall(...)`.

this is what I'm thinking about:
- get all calls of method `externalServiceCall(...)`
- from all methods extracted from previous step extract first parameter which is some enumeration implementing `Service` interface  (with this I can extract the name of service)
- continue with call hierarchy of `externalServiceCall(...)` and go upper until I find some annotation on method. In this case I'm searching for spring's [@RequestMapping](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestMapping.html)

So in final I can have something like:

- EXTERNAL_SERVICE1:
    -   GET /endpoint1
    -   POST /endpoint1/{id}
- EXTERNAL_SERVICE2:
    -   DELETE: /endpoint1
    -   DELETE: /endpoint2

this is example of how the externalService call looks like `externalServiceCall(ProductServiceEnum.GET_DETAIL, request)`


...

:D ok I know where the problem can be....I was launching that test from IntelliJ. 
However when I run this method:

```
@Test
    public void testIt() {
        MavenLauncher launcher =new MavenLauncher("/tmp/spring-websocket-rabbitmq-poc", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
        launcher.getEnvironment().setNoClasspath(true);
        launcher.buildModel();

        System.out.println(getClassPathEntries(launcher.getModelBuilder().getSourceClasspath()));
        CtModel model = launcher.getModel();

        List<CtMethod> allMethods = model.getElements(new TypeFilter<>(CtMethod.class));
        for (CtMethod<?> myMethod : allMethods) {
            CtExecutableReference execRef = myMethod.getReference();
            List<CtInvocation<?>> invocations = model.getElements(new InvocationFilter(execRef));
            System.out.println("test");
        }
    }
```

I get error:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.875 s <<< FAILURE! - in com.github.bilakpoc.spoonresttobackendmappinganalyzer.QuickTest
[ERROR] testIt(com.github.bilakpoc.spoonresttobackendmappinganalyzer.QuickTest)  Time elapsed: 0.849 s  <<< ERROR!
spoon.support.SpoonClassNotFoundException: cannot create shadow class: org.springframework.web.socket.config.annotation.AbstractWebSocketMessageBrokerConfigurer
        at com.github.bilakpoc.spoonresttobackendmappinganalyzer.QuickTest.testIt(QuickTest.java:34)
Caused by: java.lang.NoClassDefFoundError: org/springframework/messaging/simp/config/ChannelRegistration
        at com.github.bilakpoc.spoonresttobackendmappinganalyzer.QuickTest.testIt(QuickTest.java:34)
Caused by: java.lang.ClassNotFoundException: org.springframework.messaging.simp.config.ChannelRegistration
        at com.github.bilakpoc.spoonresttobackendmappinganalyzer.QuickTest.testIt(QuickTest.java:34)
```

where line 34 is this one ` List<CtInvocation<?>> invocations = model.getElements(new InvocationFilter(execRef));`. Now (as It hurts to think :D) I don't know if that's problem of some logic in code or is there any other issue.

and question: were you or anyone able to run spoon inside Intellij? It would be very helpfull to debug it in it.

--- a/src/main/java/spoon/support/reflect/reference/CtExecutableReferenceImpl.java
+++ b/src/main/java/spoon/support/reflect/reference/CtExecutableReferenceImpl.java
@@ -166,7 +166,11 @@ public class CtExecutableReferenceImpl<T> extends CtReferenceImpl implements CtE
 
 	@Override
 	public CtExecutable<T> getExecutableDeclaration() {
-		return getCtExecutable(getDeclaringType().getTypeDeclaration());
+		CtTypeReference<?> declaringType = getDeclaringType();
+		if (declaringType == null) {
+			return null;
+		}
+		return getCtExecutable(declaringType.getTypeDeclaration());
 	}
 
 	private CtExecutable<T> getCtExecutable(CtType<?> typeDecl) {

